ogr/ogrsf_frmts/miramon/mm_gdal_functions.h

```diff
@@ -0,0 +1,162 @@
+#ifndef __MM_GDAL_FUNCTIONS_H
+#define __MM_GDAL_FUNCTIONS_H
+/* -------------------------------------------------------------------- */
+/*      Constants used in GDAL and in MiraMon         */
+/* -------------------------------------------------------------------- */
+
+#ifdef GDAL_COMPILATION
+#include "mm_gdal_constants.h"       // MM_BYTE
+#include "mm_gdal_structures.h"      // struct BASE_DADES_XP
+#include "mm_gdal_driver_structs.h"  // struct MMAdmDatabase
+CPL_C_START                          // Necessary for compiling in GDAL project
+#else
+#include "mm_constants.h"                    // MM_BYTE
+#include "mm_gdal\mm_gdal_structures.h"      // struct BASE_DADES_XP
+#include "mm_gdal\mm_gdal_driver_structs.h"  // struct MMAdmDatabase
+#endif
+
+#define nullptr NULL
+
+    // Log. It should be temporal
+    extern const char *MM_pszLogFilename;
+
+#define LOG_STR(str) (Log((str), __LINE__))
+#define LOG_ACTION(action) ((void)Log(#action, __LINE__), (action))
+
+const char *Log(const char *pszMsg, int nLineNumber);
+
+// MiraMon feature table descriptors
+#define MM_MAX_IDENTIFIER_SIZE 50
+#define MM_a_WITH_GRAVE (-32)
```

Comment:
negative value, because you compare them with char ?  Beware that isn't to be portable, since char is unsigned on some platforms, especially ARM.  An alternative would be to put the unsigned values here (that is those ones + 256), and make sure everywhere you compare with those constants, you cast explicitly to (unsigned char)